docs/user-guide/petset.md

```diff
@@ -0,0 +1,370 @@
+
+nginx.default.svc.cluster.local	service = 10 50 0 web-1.ub.default.svc.cluster.local.
+nginx.default.svc.cluster.local	service = 10 50 0 web-0.ub.default.svc.cluster.local.
+```
+
+### Ordering
+
+While stable identites might decouple the physical infrastructure from units of compute, it's not sufficient to decouple system administration from application development. This is a problem the industry hasn't solved for many years, and we're probably not going to solve it now, but there are certain properties of an ordered set that we can leverage to significantly reduce the probability that someone wakes up to an alarming pager.
+
+As an example, consider a common application deployment setup: master/slave. This usually involves picking a single instance of a database as the "master", and parenting every other instance to it, as a "slave". The instructions for most master/slave setups begin with: contact a admin, and end with: run to the hills. Jokes aside, this is because deploying master/slave is a little delicate. The steps are:
+
+* Start 1 database as the master
+* Find its ip
+* Start all other databases pointed at the master ip
+
+We're intentionally ignoring the failover scenario.
+If you think about what the admin is doing, he/she is:
+
+1. Making sure the first database is started in stand alone mode, meaning it doesn't take any input about peers, and no peers are trying to join it.
+2. Waiting till it's "healthy" (maybe till it can serve reads)
+3. Checking its ip
+4. Starting one more database with the master ip as input
+5. Repeat from step 2 for each slave
+
+A meticulous admin would re-verify that the master ip hasn't changed on each iteration, and that all previous pets continue to believe they are slaves of the current master. The Pet Set controller tries to mimic these actions. It will always start one pet at a time and wait for it to enter Running and Ready before starting the next one. Instead of using ips, it provisions DNS records for each pet, so even if the IP does change the DNS name will not.
+
+There are 2 issues that make automating even a simple workflow like this tricky:
+
+* how do we pick the master? the admin picked the first peer, since we know Pets are always started in order and one-at-a-time, we can promote the first pod that doesn't find any peers (we can also use the ordinal index, but that's more brittle).
+* how do we know its ip? the admin probably ran `ip addr`. If a given Pet sees more than 1 peer it knows it isn't master. It needs to query all the others to find the master.
+
+So it's safe to say that if each Pet had a way of finding the exact list of its peers as it comes up, and that list doesn't change, and that list *includes itself*, that it can figure out what it needs to join the cluster. This leads us to the next problem, how do we wait till the list of peers includes what we know is our DNS name? (a Pet *cannot* bootstrap till this time because its peers will not be able to resolve it).
```

Comment:
Line 200 needs more work. There is probably a way to make this point more clearly.  "What it needs" is a bit vague are we just talking master selection?